boards/arm/96b_avenger96/doc/index.rst

```diff
@@ -0,0 +1,276 @@
+----------------------------------
+
+- UART_7 TX/RX/RTS/CTS : PE8/PE7/PE9/PE10 (UART console)
+- UART_4 TX/RX : PD1/PB2
+
+System Clock
+------------
+
+The Cortex®-M4 Core is configured to run at a 209 MHz clock speed.
+
+Serial Port
+-----------
+
+96Boards Avenger96 board has 3 U(S)ARTs. The Zephyr console output is assigned
+by default to the RAM console to be dumped by the Linux Remoteproc Framework
+on Cortex®-A7 core. Alternatively, Zephyr console output can be assigned to
+UART7 which is disabled by default. UART console can be enabled through
+board's device tree and 96b_avenger96_defconfig board file (or prj.conf
+project files), and will disable existing RAM console output. Default UART
+console settings are 115200 8N1.
+
+Programming and Debugging
+*************************
+
+The STM32MP157A doesn't have QSPI flash for the Cortex®-M4  and it needs to be
+started by the Cortex®-A7 core. The Cortex®-A7 core is responsible to load the
+Cortex®-M4 binary application into the RAM, and get the Cortex®-M4 out of reset.
+The Cortex®-A7 can perform these steps at bootloader level or after the Linux
+system has booted.
+
+The Cortex®-M4 can use up to 2 different RAMs. The program pointer starts at
+address 0x00000000 (RETRAM), the vector table should be loaded at this address
+These are the memory mappings for Cortex®-A7 and Cortex®-M4:
+
++------------+-----------------------+------------------------+----------------+
+| Region     | Cortex®-A7            | Cortex®-M4             | Size           |
++============+=======================+========================+================+
+| RETRAM     | 0x38000000-0x3800FFFF | 0x00000000-0x0000FFFF  | 64KB           |
++------------+-----------------------+------------------------+----------------+
+| MCUSRAM    | 0x10000000-0x1005FFFF | 0x10000000-0x1005FFFF  | 384KB          |
++------------+-----------------------+------------------------+----------------+
+| DDR        | 0xC0000000-0xFFFFFFFF |                        | up to 1 GB     |
++------------+-----------------------+------------------------+----------------+
+
+
+Refer to `stm32mp157 boot Cortex-M4 firmware`_ wiki page for instruction
+to load and start the Cortex-M4 firmware.
+
+Debugging
+=========
+
+You can debug an application using OpenOCD and GDB. The Solution proposed below
+is based on the Linux STM32MP1 SDK OpenOCD and is available only for a Linux
+environment. The firmware must first be loaded by the Cortex®-A7. Developer
+then attachs the debugger to the running Zephyr using OpenOCD.
+
+Prerequisite
+------------
+install `stm32mp1 developer package`_.
+
+1) start OpenOCD in a dedicated terminal
+
+   - Start up the  sdk environment::
+
+      source <SDK installation directory>/environment-setup-cortexa7hf-neon-vfpv4-openstlinux_weston-linux-gnueabi
+
+   - Start OpenOCD::
+
+      ${OECORE_NATIVE_SYSROOT}/usr/bin/openocd -s ${OECORE_NATIVE_SYSROOT}/usr/share/openocd/scripts -f board/stm32mp15x_dk2.cfg
```

Comment:
I thought so but it is not ;-) It should be `stm32mp15x_ev1_jlink_jtag.cfg`, since the board exposes only standard JTAG interface but I haven't tested it.